ApiBundle/Controller/GroupController.php

```diff
@@ -69,97 +48,117 @@ public function editAction(Request $request, $groupId)
             'label' => 'labels',
         );
+
         $configuration = PaginateFinderConfiguration::generateFromRequest($request, $mapping);
         $repository = $this->get('open_orchestra_user.repository.group');
         $collection = $repository->findForPaginate($configuration, $siteIds);
+
+        $nbrGroupsUsers = array();
+        if ($withCount) {
+            $filter = $collection;
+            array_walk($filter, function(&$item) {$item = $item->getId();});
+            $nbrGroupsUsers = $this->get('open_orchestra_user.repository.user')->countUserByGroup($filter);
+        }
+
         $recordsTotal = $repository->count($siteIds);
         $recordsFiltered = $repository->countWithFilter($configuration, $siteIds);
 
         $collectionTransformer = $this->get('open_orchestra_api.transformer_manager')->get('group_collection');
-        $facade = $collectionTransformer->transform($collection);
+        $facade = $collectionTransformer->transform($collection, $nbrGroupsUsers);
         $facade->recordsTotal = $recordsTotal;
         $facade->recordsFiltered = $recordsFiltered;
 
         return $facade;
     }
 
     /**
-     * @param string $groupId
+     * @param Request $request
      *
-     * @Config\Route("/{groupId}/delete", name="open_orchestra_api_group_delete")
+     * @Config\Route("/delete-multiple", name="open_orchestra_api_group_delete_multiple")
      * @Config\Method({"DELETE"})
      *
      * @return Response
      */
-    public function deleteAction($groupId)
+    public function deleteGroupsAction(Request $request)
     {
-        $group = $this->get('open_orchestra_user.repository.group')->find($groupId);
-        $dm = $this->get('object_manager');
-        $this->dispatchEvent(GroupEvents::GROUP_DELETE, new GroupEvent($group));
-        $dm->remove($group);
-        $dm->flush();
+        $format = $request->get('_format', 'json');
+        $facade = $this->get('jms_serializer')->deserialize(
+            $request->getContent(),
+            $this->getParameter('open_orchestra_api.facade.group_collection.class'),
+            $format
+        );
+        $groupRepository = $this->get('open_orchestra_user.repository.group');
+        $groups = $this->get('open_orchestra_api.transformer_manager')->get('group_collection')->reverseTransform($facade);
+        $groupIds = array();
+        foreach ($groups as $group) {
+            if ($this->isGranted(ContributionActionInterface::DELETE, $group)) {
+                $groupIds[] = $group->getId();
+                $this->dispatchEvent(GroupEvents::GROUP_DELETE, new GroupEvent($group));
+            }
+        }
+
+        $groupRepository->removeGroups($groupIds);
 
         return array();
     }
 
     /**
-     * @param string $groupId
+     * @param Request $request
      *
-     * @Config\Route("/{groupId}/duplicate", name="open_orchestra_api_group_duplicate")
+     * @Config\Route("/duplicate/datatable", name="open_orchestra_api_group_duplicate")
```

Comment:
I need to fragment to not match edit route and this duplicate is done from datatable